supertokens_python/recipe/emailpassword/emaildelivery/service/backwardCompatibility/__init__.py

```diff
@@ -32,53 +31,84 @@
+        EmailPasswordRecipe
+    from supertokens_python.recipe.emailpassword.utils import (
+        InputEmailVerificationConfig, InputResetPasswordUsingTokenFeature)
+
+
+def default_create_and_send_custom_email(
+        app_info: AppInfo) -> Callable[[User, str, Dict[str, Any]], Awaitable[None]]:
+    async def func(user: User, password_reset_url_with_token: str, _: Dict[str, Any]):
+        if ('SUPERTOKENS_ENV' in environ) and (environ['SUPERTOKENS_ENV'] == 'testing'):
+            return
+        try:
+            data = {
+                'email': user.email,
+                'appName': app_info.app_name,
+                'passwordResetURL': password_reset_url_with_token
+            }
+            async with AsyncClient() as client:
+                await client.post('https://api.supertokens.io/0/st/auth/password/reset', json=data, headers={'api-version': '0'})  # type: ignore
+        except Exception:
+            pass
+
+    return func
+
 
 class CreateAndSendCustomEmailInput:
     user: str
     email_verification_url_with_token: str
     user_context: Dict[str, Any]
 
+
+class ResetPasswordUsingTokenFeatureWrapper:
+    def __init__(self, create_and_send_custom_email: Callable[[User, str, Dict[str, Any]], Awaitable[None]]) -> None:
+        self.create_and_send_custom_email = create_and_send_custom_email
+
+
 class BackwardCompatibilityService(EmailDeliveryInterface[TypeEmailPasswordEmailDeliveryInput]):
     app_info: AppInfo
     emailVerificationBackwardCompatiblityService: EmailVerificationBackwardCompatibilityService
 
     def __init__(self,
-            app_info: AppInfo,
-            recipeInterfaceImpl: EmailPasswordRecipe, # Any,  # FIXME: Should be EmailPasswordRecipe. But leads to circular dependency
-            reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
-            email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
-        ) -> None:
+                 app_info: AppInfo,
+                 recipeInterfaceImpl: EmailPasswordRecipe,
+                 reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
+                 email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
+                 ) -> None:
         self.app_info = app_info
         self.recipeInterfaceImpl = recipeInterfaceImpl
-        self.reset_password_using_token_feature = reset_password_using_token_feature
 
-        if reset_password_using_token_feature:
-            if reset_password_using_token_feature.create_and_send_custom_email is None:
-                reset_password_using_token_feature.create_and_send_custom_email = default_create_and_send_custom_email(self.app_info)
-            self.reset_password_using_token_feature = reset_password_using_token_feature
+        rputf_create_and_send_custom_email = default_create_and_send_custom_email(self.app_info)
```

Comment:
change rputf_create_and_send_custom_email to reset_password_feature_send_email_func

Reply:
Much better. Thanks!